type-definitions/ts-tests/ordered-map.ts

```diff
@@ -1,479 +1,510 @@
+import { expect, test } from 'tstyche';
 import { OrderedMap, List } from 'immutable';
 
-{
-  // #constructor
+test('#constructor', () => {
+  expect(OrderedMap()).type.toEqual<OrderedMap<unknown, unknown>>();
 
-  // $ExpectType OrderedMap<unknown, unknown>
-  OrderedMap();
+  expect(OrderedMap<number, number>()).type.toEqual<
+    OrderedMap<number, number>
+  >();
 
-  // $ExpectType OrderedMap<number, number>
-  OrderedMap<number, number>();
+  expect(OrderedMap([[1, 'a']])).type.toEqual<OrderedMap<number, string>>();
 
-  // $ExpectType OrderedMap<number, string>
-  OrderedMap([[1, 'a']]);
+  expect(OrderedMap(List<[number, string]>([[1, 'a']]))).type.toEqual<
+    OrderedMap<number, string>
+  >();
 
-  // $ExpectType OrderedMap<number, string>
-  OrderedMap(
-    List<[number, string]>([[1, 'a']])
-  );
-
-  // $ExpectType OrderedMap<string, number>
-  OrderedMap({ a: 1 });
+  expect(OrderedMap({ a: 1 })).type.toEqual<OrderedMap<string, number>>();
 
   // No longer works in typescript@>=3.9
   // // $ExpectError - TypeScript does not support Lists as tuples
   // OrderedMap(List([List(['a', 'b'])]));
```

Comment:
Hm.. I guess type is not correctly inferred here:

<img width="877" alt="Screenshot 2024-02-29 at 11 13 08" src="https://github.com/immutable-js/immutable-js/assets/72159681/40a796d3-c61b-4d37-b6a3-bcd569fbda86">

`$ExpectError` does nothing, because there is no error anymore. The lines could be removed or I could add `expect.fail()`. Only question what is expected type? Hm.. I can't figure it out.